### PR TITLE
Fix/email validation

### PR DESCRIPTION
**Pull request by matAlmeida**
add the validation email feature using the following posts as reference:

https://help.returnpath.com/hc/en-us/articles/220560587-What-are-the-rules-for-email-address-syntax-
https://stackoverflow.com/questions/2049502/what-characters-are-allowed-in-an-email-address

### CODE REVIEW

**Comment by matAlmeida** [comment]
@hyanmandian @fxamauri

File: src/utilities/email/index.ts [hunk 1]
```diff
@@ -1,5 +1,22 @@
...
   if (!email || typeof email !== 'string') return false;
 
-  return /^[a-zA-Z0-9_.-]+@[a-zA-Z0-9-]+.[a-zA-Z0-9-.]+$/.test(email);
+  const maxRecipientLength = 64;
+  const maxDomainLength = 253;
+  const maxEmailLength = maxRecipientLength + 1 + maxDomainLength;
+
+  if (email.length > maxEmailLength) return false;
+
+  const validEmailRegex = /^([!#$%&'*+\-/=?^_`{|}~]{0,1}([a-zA-Z0-9][!#$%&'*+\-/=?^_`{|}~.]{0,1})+)@(([a-zA-Z0-9][-.]{0,1})+)([.]{1}[a-zA-Z0-9]+)$/;
```
Comment (hyanmandian) [inline src/utilities/email/index.ts]: You can move this regex to keep together with others `consts`.

File: src/utilities/email/index.ts [hunk 2]
```diff
@@ -1,5 +1,22 @@
...
+
+  const matchedEmail = validEmailRegex.exec(email);
+
+  if (!matchedEmail) return false;
+
+  const [, recipient, , domain, , topLevelDomain] = matchedEmail;
+
+  if (recipient.length > maxRecipientLength) return false;
```
Comment (hyanmandian) [inline src/utilities/email/index.ts]: Maybe extract this expression to a function...

File: src/utilities/email/index.ts [hunk 3]
```diff
@@ -1,5 +1,22 @@
...
+  if (domain.length + topLevelDomain.length > maxDomainLength) return false;
```
Comment (hyanmandian) [inline src/utilities/email/index.ts]: Maybe extract this expression to a function...

**Comment by matAlmeida** [comment]
Sorry the delay

**Pull request merged**